src/psyclone/tests/psyGen_test.py

```diff
@@ -2621,8 +2631,465 @@ def test_binaryoperation_can_be_printed():
+    f2008parser = ParserFactory().create(std="f2008")
+    fake_parent = KernelSchedule("dummy_schedule")
+    processor = Fparser2ASTProcessor()
+
+    # Test simple declarations
+    reader = FortranStringReader("integer :: l1")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("l1").name == 'l1'
+    assert fake_parent.symbol_table.lookup("l1").datatype == 'integer'
+    assert fake_parent.symbol_table.lookup("l1").shape == []
+    assert fake_parent.symbol_table.lookup("l1").access == 'local'
+
+    reader = FortranStringReader("Real      ::      l2")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    processor.process_declarations(fake_parent, [fparser2specification])
+    assert fake_parent.symbol_table.lookup("l2").name == "l2"
+    assert fake_parent.symbol_table.lookup("l2").datatype == 'real'
+    assert fake_parent.symbol_table.lookup("l2").shape == []
+    assert fake_parent.symbol_table.lookup("l2").access == 'local'
+
+    # RHS array specifications are not supported
+    reader = FortranStringReader("integer :: l1(4)")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    with pytest.raises(InternalError) as error:
+        processor.process_declarations(fake_parent, [fparser2specification])
+    assert ("Array specifications after the variable name are not "
+            "supported.") in str(error.value)
+
+    # Initialisations are not supported
+    reader = FortranStringReader("integer :: l1 = 1")
+    fparser2specification = Specification_Part.match(reader)[0][0]
+    with pytest.raises(InternalError) as error:
+        processor.process_declarations(fake_parent, [fparser2specification])
+    assert ("Array initializations on the declaration statements are not "
+            "supported.") in str(error.value)
+
+    # Char lengths are not supported
+    # TODO: It would be simpler to do just a Specification_Part.match() instead
+    # of parsing a full program, but fparser/169 needs to be fixed first.
+    reader = FortranStringReader("program dummy\ncharacter :: l*4"
+                                 "\nend program")
+    program = f2008parser(reader)
+    fparser2specification = program.content[0].content[1].content[0]
+    with pytest.raises(InternalError) as error:
+        processor.process_declarations(fake_parent, [fparser2specification])
+    assert ("Character length specifications are not "
+            "supported.") in str(error.value)
+
+
+def test_fparser2astprocessor_process_declarations_intent():
+    '''Test that process_declarations method parses multiple specifications
+    of the intent attribute.
+    '''
+    from fparser.two.parser import ParserFactory
+    from fparser.common.readfortran import FortranStringReader
+    from fparser.two.Fortran2003 import Specification_Part
+    f2008parser = ParserFactory().create(std="f2008")
```

Comment:
Rather than create a new parser for each test (that needs it) in this module, please add a fixture at the beginning of the module that just creates one on module entry.

    @pytest.fixture(scope="module")
    def f2008_parser():
        return ParserFactory().create(std="f2008")

you then provide this fixture to each test that needs it (as an argument to the test method).

Reply:
Good one